boa/src/builtins/array/mod.rs

```diff
@@ -1114,7 +1114,8 @@ impl Array {
 
     /// Initialise the `Array` object on the global object.
     #[inline]
-    pub(crate) fn init(global: &Value) -> (&str, Value) {
+    pub(crate) fn init(interpreter: &mut Interpreter) -> (&'static str, Value) {
+        let global = &interpreter.realm.global_obj;
```

Comment:
It would make the code cleaner  if we used a getter function here that would return a reference of the global object. `Interpreter::global()`, instead of `&interpreter.realm.global_obj`